projects/charger/inc/charger_fsm.h

```diff
@@ -0,0 +1,42 @@
+#pragma once
+// FSM for the charger
+//
+// Requires
+// - notify
+// - charger_pin
+// - charger_controller
+// To be initialized.
+//
+// Charger design:
+//
+// Charger Disconnected:
+// - Do nothing
+//
+// Charger Connected:
+// - Notify of connection periodically
+//
+// Charger Charging:
+// - Notify of connection periodically
+// - Charge while command is not expired and periodically publish data
+// - Exits immediately in case of fault
+//
+// Transition Table:
+// | From \ To    | Disconnected | Connected | Charging |
+// | Disconnected |              |     Y     |     N    |
+// | Connected    |       Y      |           |   G(Y)   |
+// | Charging     |       Y      |           |          |
+//
+// Y - Allowed
+// N - Not Allowed
+// G(Y) - Guarded but Allowed
+
+#include <stdbool.h>
+
+#include "charger_can.h"
+#include "event_queue.h"
+
+// Initializes the FSM |charger_can_status| is used to guard against faults.
+void charger_fsm_init(ChargerCanStatus *charger_can_status);
```

Comment:
I'm not a huge fan of passing a `ChargerCanStatus *` that I assume will magically change on CAN RX. Since it seems like you'd be getting the data from `charger_controller` anyways, I think it would make more sense to expose a function to access it or something similar.

Reply:
Done. This was primarily to make testing easier but sure.